common/persistence/cassandra/metadata_store.go

```diff
@@ -198,6 +202,110 @@ func (m *MetadataStore) UpdateNamespace(request *p.InternalUpdateNamespaceReques
 	return nil
 }
 
+func (m *MetadataStore) RenameNamespace(request *p.InternalRenameNamespaceRequest) error {
+	return m.RenameNamespace1(request)
+}
+
+// Does almost everything in one batch. The only separate thing is namespaecs_by_id update which is almost not used now.
+// Can be retried but data between steps is inconsistens (points to the name which doesn't exist).
+func (m *MetadataStore) RenameNamespace1(request *p.InternalRenameNamespaceRequest) error {
```

Comment:
i would prefer option 1

the additional namespace ID -> namespace name mapping is to ensure namespace ID being unique
the ID -> name mapping (DB table) is not used according to my memory

the ID -> name lookup is done by namespace cache 
* loading all namespace from `namespaces` table 
* index ID -> name using above records

Reply:
could you add some comment about the `namespaces_by_id` table? thanks
btw, to successfully achieve the namespace renaming functionality, we need to guarantee only IDs are used within service boundary

Reply:
I improved comment here. I can't expose `RenameNamespace` as public API for now. I added it only to support namespace deletion.